### PR TITLE
Adding support for uploading files from react native

### PR DESCRIPTION
**Pull request by wmathurin**
Goes hand in hand with: https://github.com/forcedotcom/SalesforceMobileSDK-Shared/pull/343

### CODE REVIEW

**Comment by wmathurin** [comment]
New sample app demonstrating file upload here: https://github.com/wmathurin/MyUserPicReactNative

File: libs/SalesforceReact/SalesforceReact/Classes/SFNetReactBridge.m [hunk 1]
```diff
@@ -50,13 +54,30 @@ @implementation SFNetReactBridge
...
+    if (fileParams) {
+        // File params expected to be of the form:
+        // {<fileParamNameInPost>: {fileMimeType:<someMimeType>, fileUrl:<fileUrl>, fileName:<fileNameForPost>}}
+        for (NSString* fileParamName in fileParams) {
+            NSDictionary* fileParam = fileParams[fileParamName];
+            NSString* fileMimeType = [fileParam nonNullObjectForKey:kFileMimeType];
+            NSString* fileUrl = [fileParam nonNullObjectForKey:kFileUrl];
+            NSString* fileName = [fileParam nonNullObjectForKey:kFileName];
+            NSData* fileData = [NSData dataWithContentsOfURL:[NSURL URLWithString:fileUrl]];
+            [request addPostFileData:fileData paramName:fileParamName fileName:fileName mimeType:fileMimeType];
```
Comment (kchitalia) [inline libs/SalesforceReact/SalesforceReact/Classes/SFNetReactBridge.m]: What happens if for some reason the request fails?

Comment (wmathurin) [inline libs/SalesforceReact/SalesforceReact/Classes/SFNetReactBridge.m]: If the send request fails, we send back the error to javascript (see line 83 below)

Comment (kchitalia) [inline libs/SalesforceReact/SalesforceReact/Classes/SFNetReactBridge.m]: Got it.

**Comment by kchitalia** [comment]
:+1:

**Pull request merged**